src/service/git/github-api.ts

```diff
@@ -106,8 +113,31 @@ export class GithubAPIService {
         throw new NotFoundError();
       }
     } catch (err) {
-      this.logger.error(`Error getting fork name for ${targetOwner}/${repo} where owner is ${sourceOwner}`);
+      this.logger.error(
+        this.getErrorMessage(
+          err, 
+          `Error getting fork name for ${targetOwner}/${repo} where owner is ${sourceOwner}`
+        )
+      );
       throw err;
     }
   }
+
+  private getErrorMessage(err: unknown, msg: string): string {
+    if (err instanceof RequestError) {
+      let reason;
+      switch (err.status) {
+        case 401:
+          reason = "Failed to authenticate with provided token, please use -token argument to provide a new one. You can also check your GITHUB_TOKEN environment variable and check whether the provided token is still valid.";
+        case 404:
+          reason = "Failed to fetch GitHub URL, please check if the URL used in -url argument is valid and if the token you are using have permissions to access it.";
+        case 403:
+          reason = "Failed to fetch resource. Either your github token does not have access to the requested resource or you have reached your github api rate limit.";
```

Comment:
```suggestion
          reason = "Failed to fetch resource. Either your github token does not have access to the requested resource or you have reached your github api rate limit.";
          break;
```

Reply:
don't need a break for the last case no?

Reply:
it is a good pattern to add a break per case and also a default section, so I would recommend to do so, the default case is basically empty

Reply:
ok will do